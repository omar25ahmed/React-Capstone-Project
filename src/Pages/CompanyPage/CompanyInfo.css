ul {
  padding: 0;
  margin: 0;
  list-style: none;
}

li {
  height: 50px;
  padding: 15px;
}
