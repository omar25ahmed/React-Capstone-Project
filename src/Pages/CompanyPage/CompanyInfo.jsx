import React from 'react';
import PropTypes from 'prop-types';
import './CompanyInfo.css';

function CompanyInfo({ company }) {
  const {
    marketCap, sector, industry, price, exchange, exchangeShortName, isActivelyTrading,
  } = company;
  const nf = new Intl.NumberFormat('en-US');
  return (
    <ul data-testid="company info">
      <li className="d-flex justify-content-between">
        <span>Market Cap</span>
        {nf.format(marketCap)}
        {' '}
        $
      </li>
      <li className="d-flex justify-content-between">
        <span>Sector</span>
        {sector}
      </li>
      <li className="d-flex justify-content-between">
        <span>Industry</span>
        {industry}
      </li>
      <li className="d-flex justify-content-between">
        <span>Price of the company</span>
        {price}
        {' '}
        M
      </li>
      <li className="d-flex justify-content-between">
        <span>Exchange</span>
        {exchange}
      </li>
      <li className="d-flex justify-content-between">
        <span>Exchange shortcut</span>
        {exchangeShortName}
      </li>
      <li className="d-flex justify-content-between">
        <span>Trading status</span>
        {isActivelyTrading ? 'active' : 'not active'}
      </li>
    </ul>
  );
}

export default CompanyInfo;

CompanyInfo.defaultProps = {
  company: {
    companyName: '',
    symbol: '',
  },
};

CompanyInfo.propTypes = {
  company: PropTypes.shape({
    marketCap: PropTypes.number,
    sector: PropTypes.string,
    industry: PropTypes.string,
    price: PropTypes.number,
    exchange: PropTypes.string,
    exchangeShortName: PropTypes.string,
    isActivelyTrading: PropTypes.bool,
  }),
};
