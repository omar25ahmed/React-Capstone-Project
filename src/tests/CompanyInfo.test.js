import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/Store';
import CompanyInfo from '../Pages/CompanyPage/CompanyInfo';

describe('testing Home component', () => {
  it('snapshot for the Company Info ', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <CompanyInfo />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Company Info component renders correctly', () => {
    render(
      <Provider store={store}>
        <CompanyInfo />
      </Provider>,
    );
    expect(screen.getByTestId('company info')).toBeInTheDocument();
  });

  it('should GET data', async () => {
    window.fetch = jest.fn();
    window.fetch.mockResolvedValueOnce({
      json: async () => [{ symbol: '', companyName: '' }],
    });
    render(
      <Provider store={store}>
        <CompanyInfo />
      </Provider>,
    );
    const listItemElements = await screen.findAllByRole('listitem');
    expect(listItemElements).not.toHaveLength(0);
  });
});
