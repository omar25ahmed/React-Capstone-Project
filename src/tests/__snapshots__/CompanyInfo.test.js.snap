// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home component snapshot for the Company Info  1`] = `
<ul
  data-testid="company info"
>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Market Cap
    </span>
    NaN
     
    $
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Sector
    </span>
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Industry
    </span>
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Price of the company
    </span>
     
    M
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Exchange
    </span>
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Exchange shortcut
    </span>
  </li>
  <li
    className="d-flex justify-content-between"
  >
    <span>
      Trading status
    </span>
    not active
  </li>
</ul>
`;
